framework/src/controller/bus.js

```diff
@@ -0,0 +1,68 @@
+const Promise = require('bluebird');
+const { EventEmitter2 } = require('eventemitter2');
+const Action = require('./action');
+
+module.exports = class Bus extends EventEmitter2 {
+	constructor(controller, options) {
+		super(options);
+		this.controller = controller;
+
+		// Hash map used instead of arrays for performance.
+		this.actions = {};
+		this.events = {};
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	async setup() {
+		// Place holder for RPC server connection
+		return Promise.resolve();
+	}
+
+	// eslint-disable-next-line no-unused-vars
+	async registerChannel(moduleAlias, events, actions, options) {
+		events.forEach(e => {
+			const eventName = `${moduleAlias}:${e}`;
+			if (this.events[eventName]) {
+				throw new Error(`Event "${eventName}" already registered with bus.`);
+			}
+			this.events[eventName] = true;
+		});
+
+		actions.forEach(a => {
+			const actionName = `${moduleAlias}:${a}`;
+			if (this.actions[actionName]) {
+				throw new Error(`Action "${actionName}" already registered with bus.`);
+			}
+			this.actions[actionName] = true;
+		});
+	}
+
+	invoke(actionData) {
+		const action = Action.deserialize(actionData);
+
+		if (action.module === 'lisk') {
```

Comment:
Why is 'lisk' value hardcoded? Should be passed by configuration as app name.

Reply:
That's an identifier for controller, yes it most be moved to a constant and will do it. First we need to define what is app name is defining, if that identifier remains same among networks of same app then we can use it. If not then we have to stick with one.